nengo/builder/connection.py

```diff
@@ -169,7 +171,7 @@ def get_prepost_signal(is_pre):
     # Sample transform if given a distribution
     transform = (conn.transform.sample(conn.size_out, conn.size_mid, rng=rng)
                  if isinstance(conn.transform, Distribution) else
-                 conn.transform)
+                 np.array(conn.transform))
```

Comment:
The code used to try to be super smart about this, and only made the copy if we were not in that case where we were solving for decoders. But it made things a bit more complicated to read, and I also wanted to avoid making a copy of the sampled distribution, since that's extra computation (though arguably memory savings are more important). And then I realized that if we're storing it in BuiltConnection, it would be nice to have a copy anyway, and so that convinced me that it's fine to just make a copy right off the bat. A readonly view is a possibility, though the thing I always worry about is a user creates an array, sets it to be the transform, then modifies the original array. There's no way that we can set the user's view on their array to be readonly, so they'll always be able to do this. But maybe that's their own damn fault? There are definitely arguments for saving memory, especially when these transforms are in neuron->neuron connections and therefore possibly very large. Could we have an rc setting that copies by default, but the user can advise to not copy if they need the memory?